client/lib/pages/plant-list.dart

```diff
@@ -1,138 +1,148 @@
+import 'dart:async';
+
+import 'package:cached_network_image/cached_network_image.dart';
 import 'package:flutter/material.dart';
+import 'package:plantexpert/api/ApiConnectionException.dart';
 import 'package:plantexpert/api/Plant.dart';
-import 'package:plantexpert/api/User.dart';
+import 'package:plantexpert/api/UserPlant.dart';
+import 'package:plantexpert/Utility.dart';
+import 'package:plantexpert/widgets/PlantListItem.dart';
+import 'package:plantexpert/widgets/stateful-wrapper.dart';
 
 import '../MenuNavigation.dart';
 
-class PlantList extends StatelessWidget {
+Future<CachedNetworkImage> getUserPlantImage(UserPlant userPlant) async {
+  try {
+    CachedNetworkImage img = await PlantenApi.instance.connection.fetchCachedPlantImage(userPlant);
+    return img;
+  }
+  on ApiConnectionException catch (e) {
+    print("API image fetch error: " + e.toString());
+  }
+  on InvalidCredentialsException catch (e) {
+    print('Not logged in.');
+  }
+  on NetworkImageLoadException catch (e) {
+    print('There was a problem loading `${userPlant.imageName}`.');
+  }
+
+  return null;
+}
+
+class PlantList extends StatefulWidget {
+  PlantList({Key key}) : super(key: key);
+
+  @override
+  _PlantListState createState() => _PlantListState();
+
+}
+
+class _PlantListState extends State<PlantList> {
+  String failedFetchingPlants;
+
+  // save the fetched plants so they don't have to get fetched multiple times in a row 
+  List<PlantListItem> plantListItems;
+
+  void initState() {
+    super.initState();
+    _fetchUserPlants();
+  }
+
+  Future<void> _reloadUserPlants() async {
+    return _fetchUserPlants();
+  }
+
+  Future<List<Plant>> _fetchPlants() async {
+    return PlantenApi.instance.connection.fetchPlants();
+  }
+
+  void _fetchUserPlants() async {
+    failedFetchingPlants = null;
+
+    try {
+      List<UserPlant> pli = await PlantenApi.instance.connection.fetchUserPlants();
```

Comment:
This api call throws a `StatusCodeException` with status  code 404 when opening the plant-list page when logged in in an account without user plants. This exception is not handled so the page keeps 'loading'.